Vic2ToHoI4/Source/HOI4World/Diplomacy/Faction.h

```diff
@@ -20,17 +20,20 @@ class Faction
 	public:
 		Faction(
 			std::shared_ptr<Country> leader,
-			std::vector<std::shared_ptr<Country>> members
-		): factionLeader(std::move(leader)), factionMembers(std::move(members)) {}
+			std::vector<std::shared_ptr<Country>> members, 
+			std::optional<std::string> name = ""
```

Comment:
Don't do a default assignment here. As-is, it will circumvent the optional part. If you leave that out, it'll default to std::nullopt.

Reply:
Unfortunately, factions are being used in the war creator as well for a different purpose. however, I can change the default assignment to std::nullopt without breaking anything.